### PR TITLE
feat(cli): add --dry-run to uninstall command

### LINKED ISSUE #1061: `glasskube uninstall` should support `--dry-run`
**Is your feature request related to a problem? Please describe.**
Just as many other commands, `uninstall` should have a `--dry-run` option to simulate package uninstallation. 

**Additional context**
#629

### PR DESCRIPTION
**Pull request by hanshal101**
Closes #1061 
Fixes #1061 

## 📑 Description
This PR adds the `DryRun` options to the uninstall command

## ✅ Checks
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required

### CODE REVIEW

**Review by christophenne — DISMISSED** [review]
@hanshal101 please test your implementation again, when running `glasskube uninstall cyclops --dry-run`, the command never finishes or seems to wait forever.

**Comment by hanshal101** [comment]
Hey @christophenne sorry for the late update, this error was caused in the `awaitDeletion` function, here I have passed the `isDryRun` bool to check if its true then return nil.

File: pkg/uninstall/uninstall.go [hunk 1]
```diff
@@ -30,28 +30,31 @@ func (obj *uninstaller) WithStatusWriter(sw statuswriter.StatusWriter) *uninstal
...
 	obj.status.Start()
 	defer obj.status.Stop()
-	err := obj.delete(ctx, pkg)
+	err := obj.delete(ctx, pkg, isDryRun)
 	if err != nil {
 		return err
 	}
-	return obj.awaitDeletion(ctx, pkg)
+
+	return obj.awaitDeletion(ctx, pkg, isDryRun)
```
Comment (kosmoz) [inline pkg/uninstall/uninstall.go:41]:
```suggestion
	if isDryRun {
		return nil
	} else {
		return obj.awaitDeletion(ctx, pkg)
	}
```

File: pkg/uninstall/uninstall.go [hunk 2]
```diff
@@ -64,7 +67,11 @@ func (uninstaller *uninstaller) delete(ctx context.Context, pkg ctrlpkg.Package)
 	}
 }
 
-func (obj *uninstaller) awaitDeletion(ctx context.Context, pkg ctrlpkg.Package) error {
+func (obj *uninstaller) awaitDeletion(ctx context.Context, pkg ctrlpkg.Package, isDryrun bool) error {
+	if isDryrun {
+		return nil
+	}
+
```
Comment (kosmoz) [inline pkg/uninstall/uninstall.go:74]:
```suggestion
func (obj *uninstaller) awaitDeletion(ctx context.Context, pkg ctrlpkg.Package) error {
```

**Review by kosmoz — CHANGES_REQUESTED** [review]
I think it would be better to keep the logic out of the await function (see suggestions below).

Also, I think there should be some feedback that dry run mode is enabled. See for example the configure command: https://github.com/hanshal101/glasskube/blob/26abda901fab05ceee737a99c710997f48b6c453/cmd/glasskube/cmd/configure.go#L51-L53

**Comment by kosmoz** [comment]
> Also, I think there should be some feedback that dry run mode is enabled. See for example the configure command: https://github.com/hanshal101/glasskube/blob/26abda901fab05ceee737a99c710997f48b6c453/cmd/glasskube/cmd/configure.go#L51-L53

Please add some feedback that dry-run is enabled.

**Comment by hanshal101** [comment]
Code updated !!!

File: cmd/glasskube/cmd/uninstall.go [hunk 3]
```diff
@@ -66,14 +67,18 @@ var uninstallCmd = &cobra.Command{
 			}
 		}
 
+		if uninstallCmdOptions.DryRun {
+			fmt.Fprintln(os.Stderr, "🔎 Dry-run mode is enabled. Nothing will be changed.")
+		}
+
```
Comment (kosmoz) [inline cmd/glasskube/cmd/uninstall.go:73]: Please move this so that it is shown before the uninstall details and confirmation prompt

**Review by kosmoz — APPROVED** [review]

**Pull request merged**